packages/vite-app-ts/src/config/web3ModalConfig.ts

```diff
@@ -8,66 +8,102 @@ export const web3ModalConfigKeys = {
-      name: 'Coinbase',
-      description: 'Connect to your Coinbase Wallet (not coinbase.com)',
-    },
-    package: walletLinkProvider,
-    connector: async (provider: any, _options: any) => {
-      await provider.enable();
-      return provider;
-    },
-  };
+    // Coinbase walletLink init
+    const walletLink = new WalletLink({
+      appName: 'coinbase',
+    });
+    // WalletLink provider
+    const walletLinkProvider = walletLink.makeWeb3Provider(`https://mainnet.infura.io/v3/${INFURA_ID}`, 1);
 
-  //network: 'mainnet', // Optional. If using WalletConnect on xDai, change network to "xdai" and add RPC info below for xDai chain.
-  const walletConnectEthereum = {
-    package: WalletConnectProvider,
-    options: {
-      bridge: 'https://polygon.bridge.walletconnect.org',
-      infuraId: INFURA_ID,
-      rpc: {
-        1: `https://mainnet.infura.io/v3/${INFURA_ID}`,
-        42: `https://kovan.infura.io/v3/${INFURA_ID}`,
-        100: 'https://dai.poa.network',
+    const coinbaseWalletLink = {
+      display: {
+        logo: 'https://play-lh.googleusercontent.com/PjoJoG27miSglVBXoXrxBSLveV6e3EeBPpNY55aiUUBM9Q1RCETKCOqdOkX2ZydqVf0',
+        name: 'Coinbase',
+        description: 'Connect to your Coinbase Wallet (not coinbase.com)',
       },
-    },
-  };
+      package: walletLinkProvider,
+      connector: async (provider: any, _options: any) => {
+        await provider.enable();
+        return provider;
+      },
+    };
+    providerOptions[web3ModalConfigKeys.coinbaseKey] = coinbaseWalletLink;
+  } catch (e) {
+    console.log('Failed to load config for Coinbase WalletLink: ', e);
+  }
 
-  // const torus = {
-  //   package: Torus,
-  //   options: {
-  //     networkParams: {
-  //       host: 'https://localhost:8545',
-  //       chainId: 1337,
-  //       networkId: 1337, // optional
-  //     },
-  //     config: {
-  //       buildEnv: 'development',
-  //     },
-  //   },
-  // };
+  // === WALLETCONNECT
+  try {
+    const WalletConnectProvider = (await import('@walletconnect/ethereum-provider')).default;
+    const walletConnectEthereum = {
+      package: WalletConnectProvider,
+      options: {
+        bridge: 'https://polygon.bridge.walletconnect.org',
+        infuraId: INFURA_ID,
+        rpc: {
+          1: `https://mainnet.infura.io/v3/${INFURA_ID}`,
+          42: `https://kovan.infura.io/v3/${INFURA_ID}`,
+          100: 'https://dai.poa.network',
+        },
+      },
+    };
+    providerOptions.walletconnect = walletConnectEthereum;
+  } catch (e) {
+    console.log('Failed to load config for WalletConnect: ', e);
+  }
+
+  // === AUTHEREUM
+  try {
+    const Authereum = (await import('authereum')).default;
+    providerOptions.authereum = {
+      package: Authereum,
+    };
+  } catch (e) {
+    console.log('Failed to load config for Authereum: ', e);
+  }
 
+  // === LOCALHOST STATIC

```

Comment:
should we wrap this in a try catch too?

Reply:
Yes, just updated.

My initial thought was that, those two imports being for es6 modules there should be no issue. Atm it works, but it's safer to wrap those two.

And definitely looks prettier in terms of pattern consistency :)

Reply:
BTW have I got this right - rollup has issues with commonJS modules, but not with es6 ? 

I mean I looked specifically at the .js files referenced by the imports, and they were ES6 syntax.

Wanna take away as much as possible from this story in terms of learning. Been a wild couple of hours, challenge got me hooked.

Reply:
in my tests it seems to crash on nodejs utils.  This was shimmed as nodejs modules are not available with browser but lots of web3 libs reference them.  specifically util.inspect.

Reply:
you can test this locally by running `yarn build --watch`  and `yarn serve --host` to run it in production mode locally